pkg/pidfile/cleanup_test.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !privileged_tests
+
+package pidfile
+
+import (
+	"sync"
+
+	. "gopkg.in/check.v1"
+)
+
+func (pfs *PidfileTestSuite) TestHandleCleanup(c *C) {
```

Comment:
I'm not sure I fully understand the concurrent usage of `HandleCleanup()`, it would be good to see some unit tests of that functionality too.

Reply:
I was under the impression that multiple callers could queue up cleanup functions via `DeferTerminationCleanupFunction()` with the same WaitGroup/chan, and this could cause some race condition or unexpected behaviour. With the newest version in the tree, I no longer see any concerns with this.